frontend/tests/console/topics/topic-messages-production.spec.ts

```diff
@@ -0,0 +1,254 @@
+// spec: specs/topics.md
+// seed: tests/seed.spec.ts
+
+import { expect, test } from '@playwright/test';
+
+import { TopicPage } from '../utils/TopicPage';
+
+test.use({
+  permissions: ['clipboard-write', 'clipboard-read'],
+});
+
+test.describe('Produce Messages', () => {
+  test('should produce simple text message', async ({ page }) => {
+    const topicName = `text-message-${Date.now()}`;
+    const messageContent = 'Hello Redpanda Console';
+
+    const topicPage = new TopicPage(page);
+    await topicPage.createTopic(topicName);
+    await topicPage.produceMessage(topicName, messageContent);
+
+    await test.step('Verify message in messages tab', async () => {
+      await page.goto(`/topics/${topicName}`);
+      await expect(page.getByText(messageContent)).toBeVisible();
+    });
+
+    await topicPage.deleteTopic(topicName);
+  });
+
+  test.skip('should produce message with key', async ({ page }) => {
+    const topicName = `keyed-message-${Date.now()}`;
+    const messageKey = 'user-123';
+    const messageValue = 'User data for user 123';
+
+    const topicPage = new TopicPage(page);
+    await topicPage.createTopic(topicName);
+
+    await test.step('Produce message with key', async () => {
+      await page.goto(`/topics/${topicName}/produce-record`);
+
+      // Wait for page to load
+      await expect(page.getByTestId('produce-button')).toBeVisible();
+
+      // Find and fill key editor
+      const keyEditor = page.getByTestId('produce-key-editor').locator('.monaco-editor').first();
+      await keyEditor.click();
+      await page.keyboard.insertText(messageKey);
+
+      // Find and fill value editor
+      const valueEditor = page.getByTestId('produce-value-editor').locator('.monaco-editor').first();
+      await valueEditor.click();
+      await page.keyboard.insertText(messageValue);
+
+      // Produce message
+      await page.getByTestId('produce-button').click();
+
+      // Verify message was produced - page should show the message
+      await expect(page.getByRole('cell', { name: new RegExp(messageValue, 'i') }).first()).toBeVisible({
+        timeout: 10_000,
+      });
```

Comment:
This seems unstable, with the timeout and getting cell role. I think we need a test ID here